### ISSUE TITLE
Create job application

### ISSUE THREAD

**Issue by epintozzi** [turingschool/tracker-crm#5]
A new job application should include:

- [x] *Position Title
- [x] *Application URL- Link to posting
- [x] *Job description- Text box for pasting in the entirety of the job posting (in case it gets taken down and we can't use the link anymore)
- [x] *Company (dropdown which should link to an existing company resource in the database)
- [x] Application date (date applied) (can be left blank if not yet applied)
- [x] *Status (drop down with options: Not yet applied, Submitted, Phone Screen, Interview, Code Challenge, Offer, Rejected)
- [x] Contact information (Form section built, functionality will be in separate ticket)
- [x] Notes section

Required fields indicated with a * above

## Acceptance Criteria

- [x] Above features for new job application
- [x] save button makes patch request to db
- [x] Error message capability for not filling in required fields
- [x] Cypress Testing

![Image](https://github.com/user-attachments/assets/bddc6383-b906-4522-b265-25dd99439c12)

**Comment by jchirch** [comment]
@epintozzi 
Are you able to split this ticket to make an additional backend ticket?
There's some backend work I need to complete before proceeding with the ticket as it currently stands.
I need to create a new migration to add a "job application" table to the database, seed the database, and then add CRUD functionality to create that job application.

**Comment by epintozzi** [comment]
Yes I will create the issue. I also think I fixed the permissions problem so you all will be able to make these kinds of issues and edits yourselves going forward.
